script.js

```diff
@@ -0,0 +1,165 @@
+const toDoForm = document.querySelector("form");
+const toDoInput = toDoForm.querySelector(".input-text");
+const toDoButton = toDoForm.querySelector(".input-button");
+
+const TODO_DB = "todo";
+let toDoList = [];
+
+// localStorage 모델: 웹이 로드 될 때 기존 투두리스트를 가져오기 위함
+const localStorageModel = {
+    // localStorage에 저장
+    save: () => {
+        localStorage.setItem(TODO_DB, JSON.stringify(toDoList));
+    },
+
+    // localStorage에서 로드
+    load: () => {
+        const loaded = localStorage.getItem(TODO_DB);
+        if (loaded !== null) {
+            const parsed = JSON.parse(loaded);
+            toDoList = parsed;
+        }
+    }
+}
+
+// List 모델
+const list = {
+    // 개수 계산
+    getCount: () => {
+        let pending = 0;
+        toDoList.forEach((item) => {
+            if (!item.done) {
+                pending++;
+            }
+        })
+        return {
+            pending: pending,
+            done: toDoList.length - pending
+        }
+    },
+
+    // list 대기중 완료 토글
+    updateList: (value, isDone) => {
```

Comment:
객체에 바인딩 된 메소드명은 가능하다면 상위 객체의 이름을 생략하는 경우가 많습니다. `updateList`메소드의 경우, 상위 객체는 `list`죠? 이미 상위 객체에 `list`라는 단어가 들어갔으니, updateList 메소드는 `list.updateList(value, isDone)`처럼 사용되게 되므로 불필요한 이름의 중복이 일어나 가독성을 떨어뜨립니다.

```suggestion
    update(value, isDone) {
```
이렇게 하는 것이 바람직해 보입니다!

그리고, update라는 말이 상황에 따라 매우 모호할 수 있습니다. `toggleDoneStatus`, `setDoneStatus` 처럼 함수의 기능을  명확하게 반영하는 이름을 짜는 것이 좋지 않을까 싶네요. 개인적으로는 `getDoneStatus`, `toggleDoneStatus`로 바꾸어서 함수의 인자를 최대한 줄이는 방식이 좋아 보입니다. 이를 반영한다면

```javascript
    getDoneStatus(todoText) { // value보단 text가 적절해보입니다. 투두 객체의 완료여부 속성은 'text'로 되어있기 때문이죠.
        const targetTodo = toDoList.find(todo => todo.text === todoText)
        return targetTodo.done
    },
    toggleDoneStatus(todoText) {
        toDoList.forEach(todo => {
            if (todo.text === todoText) {
                todo.done = !todo.done
            }
        }
    }

```

Reply:
네이밍에 대해서 많이 고민해봐야겠네요! update라는 말은 정말 상황에 따라 모호해 보이네요 ㅠㅠ 
근데 혹시! 함수의 인자를 줄이는 방식이 왜 좋아보이는지에 대해 여쭤봐도 될까요? ㅎㅎ 저는 최대한 함수인자 내에 많은 것을 담으려고 노력합니다..! 함수를 두개로 쪼갤 수 있는 것을 한개로 만들 수 있으니까요!

Reply:
[요 링크](https://dev.to/danialmalik/a-brief-guide-to-clean-code-functions-104h)를 참고해보시면 좋을 것 같아요 ㅎㅎ

인자에 관해선 아래 설명을 참조해주세요 ㅎㅎ
<img width="739" alt="Screen Shot 2020-09-20 at 6 51 59 PM" src="https://user-images.githubusercontent.com/42387219/93708576-5e96b500-fb72-11ea-83ad-5571be6f9dd0.png">

코딩에 정해진 정답은 없지만, 적어도 함수 인자를 가능한 한 줄이자라는 말에 대해 저자가 제시하는 이유는 나름대로 이유가 있어 보여요 ㅎㅎ 한번 저 아티클을 읽어보시면 좋을 것 같아요!

Reply:
언어마다 특성이 다르군요!
전 인자를 이용해서 함수명을 네이밍할 수 있는 Swift언어의 자유로움때문에 그렇게 생각했어요. 감사합니다.
<img width="500" alt="image" src="https://user-images.githubusercontent.com/56102421/93709232-a79d3800-fb77-11ea-8be2-008443327967.png">

Reply:
오 스위프트에는 이런게 있구나 ㅎㅎ 좋은 정보 공유 감사해요 ㅎㅎ